pynamodb/models.py

```diff
@@ -1198,34 +1248,49 @@ def _serialize(self, attr_map=False, null_check=True):
         attrs = {attributes: {}}
         for name, attr in self._get_attributes().aliased_attrs():
             value = getattr(self, name)
-            if value is None:
-                if attr.null:
-                    continue
-                elif null_check:
-                    raise ValueError("Attribute '{0}' cannot be None".format(attr.attr_name))
             if isinstance(value, MapAttribute):
                 if not value.validate():
                     raise ValueError("Attribute '{0}' is not correctly typed".format(attr.attr_name))
                 value = value.get_values()
-            serialized = attr.serialize(value)
-            if serialized is None:
+
+            serialized = Model._serialize_value(attr, value, null_check)
+            if NULL in serialized:
                 continue
+
             if attr_map:
-                attrs[attributes][attr.attr_name] = {
-                    ATTR_TYPE_MAP[attr.attr_type]: serialized
-                }
+                attrs[attributes][attr.attr_name] = serialized
             else:
                 if attr.is_hash_key:
-                    attrs[HASH] = serialized
+                    attrs[HASH] = list(serialized.values())[0]
```

Comment:
I replaced `serialized = attr.serialize(value)` above with `_serialize_value()` in order to avoid double call of `attr.serialize()` you did not like.

Since `_serialize_value()` returns value in the DynamoDB serialization format (ie. `{TYPE: VALUE}`), I need to extract the value here. 

List conversion is done because Python 3 the does not support indexing of the type returned by `values()`